scapy/contrib/pnio.py

```diff
@@ -41,37 +47,293 @@
     0xFF41:"PTCP-DelayResPDU-followup",
     0xFF42:"PTCP-DelayFuResPDU",
     0xFF43:"PTCP-DelayResPDU",
-    }
-for i in xrange(0x0100, 0x1000):
-    PNIO_FRAME_IDS[i] = "RT_CLASS_3"
-for i in xrange(0x8000, 0xC000):
-    PNIO_FRAME_IDS[i] = "RT_CLASS_1"
-for i in xrange(0xC000, 0xFC00):
-    PNIO_FRAME_IDS[i] = "RT_CLASS_UDP"
-for i in xrange(0xFF80, 0xFF90):
-    PNIO_FRAME_IDS[i] = "FragmentationFrameID"
+}
+
+
+def i2s_frameid(x):
+    if x in PNIO_FRAME_IDS:
+        return PNIO_FRAME_IDS[x]
+    elif 0x0100 <= x < 0x1000:
+        return "RT_CLASS_3 (%4x)" % x
+    elif 0x8000 <= x < 0xC000:
+        return "RT_CLASS_1 (%4x)" % x
+    elif 0xC000 <= x < 0xFC00:
+        return "RT_CLASS_UDP (%4x)" % x
+    elif 0xFF80 <= x < 0xFF90:
+        return "FragmentationFrameID (%4x)" % x
+    return x
+
+
+def s2i_frameid(x):
+    try:
+        idx = PNIO_FRAME_IDS.values().index(x)
+        return PNIO_FRAME_IDS.keys()[idx]
+    except ValueError:
+        pass
+    if x == "RT_CLASS_3":
+        return 0x0100
+    elif x == "RT_CLASS_1":
+        return 0x8000
+    elif x == "RT_CLASS_UDP":
+        return 0xC000
+    elif x == "FragmentationFrameID":
+        return 0xFF80
+    return x
 
 #################
 ## PROFINET IO ##
 #################
 
-class ProfinetIO(Packet):
+class ProfinetIO(scapy.packet.Packet):
     """Basic PROFINET IO dispatcher"""
-    fields_desc = [XShortEnumField("frameID", 0, PNIO_FRAME_IDS)]
-    overload_fields = {
-        Ether: {"type": 0x8892},
-        UDP: {"dport": 0x8892},
-        }
+    fields_desc = [
+        scapy.fields.XShortEnumField("frameID", 0, (i2s_frameid, s2i_frameid))
+    ]
 
     def guess_payload_class(self, payload):
         # For frameID in the RT_CLASS_* range, use the RTC packet as payload
-        if (self.frameID >= 0x0100 and self.frameID < 0x1000) or \
-                (self.frameID >= 0x8000 and self.frameID < 0xFC00):
-            from scapy.contrib.pnio_rtc import PNIORealTime
-            return PNIORealTime
+        if (0x0100 <= self.frameID < 0x1000) or (0x8000 <= self.frameID < 0xFC00):
+            return PNIORealTimeCyclicPDU
+        return super(ProfinetIO, self).guess_payload_class(payload)
+
+scapy.packet.bind_layers(scapy.layers.l2.Ether, ProfinetIO, type=0x8892)
```

Comment:
I understand what you mean, but i disagree: you are not using the pythonish way of importing...

Let me explain: Python want to be fast. So why would it import stuff that he doesn't need ?
In your example, does `test.py` requires the function `tutu` to work ? The answer is no: only `def toto()` requires it... 

So instead of this:
```
from test2 import tutu
def toto():
  print 'toto'
  tutu()
```
You should have 
```
def toto():
  from test2 import tutu
  print 'toto'
  tutu()
```
As Python can import anything at anytime it wants to, unlike other languages like C, Java...